### PR TITLE
First attempt

### PR DESCRIPTION
**Pull request by Kevin-J-Harrison**

### CODE REVIEW

File: app/src/main/java/com/example/kharrison/my_application/CalcFunctions.kt [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package com.example.kharrison.my_application
+
+
+class CalcFunctions {
+
+   fun add(leftNum : Int, rightNum  : Int) = leftNum + rightNum
+   fun subtract(leftNum : Int, rightNum : Int) = leftNum - rightNum
+   fun multiply(leftNum : Int, rightNum : Int) = leftNum * rightNum
+   fun divide(leftNum : Int, rightNum : Int) = leftNum / rightNum
```
Comment (cwbowron) [inline app/src/main/java/com/example/kharrison/my_application/CalcFunctions.kt:9]: Kotlin does have named functions for the basic operators, so you *might* be able to just use method references to those:  `Int.plus()`, `Int.minus()`, `Int.times()` and `Int.divi()`

File: app/src/main/java/com/example/kharrison/my_application/CalculatorController.kt [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+      } else {
+         rightHandSide += num
+      }
+      addToInputString(num)
+      invalidateControls()
+   }
+
+   private fun clickOperator(operator : Int) {
+      isOperationSelected = true
+      when (operator) {
```
Comment (cwbowron) [inline app/src/main/java/com/example/kharrison/my_application/CalculatorController.kt:89]: You might consider just doing a static map that maps between the enum and the symbol. Then you could something like....

    selectedOperator = operator
    addToInputString( operatorMap[operator] )

Or you could make use real enums for `ADDITION`, etc and have them define the strings. Then you could something like...

    selectedOperator = operator
    addToInputString( operator.symbol )

Comment (cwbowron) [reply app/src/main/java/com/example/kharrison/my_application/CalculatorController.kt:89]: enum {
        ADDITION( "+", Int::plus),
        SUBTRACTION("-", Int::minus),
        ....
     }

**Comment by cwbowron** [comment]
If you only have one dpi image and if it's not necessarily scaled for that dpi and you are just scaling to fit on display, you might consider just putting them in `drawable_nodpi/` or the `mipmap_nodpi`

File: app/src/main/java/com/example/kharrison/my_application/CalculatorController.kt [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+         DIVISION -> division(leftNum, rightNum)
+      }
+      invalidateControls()
+   }
+
+   private fun add(leftNum : Int, rightNum : Int) {
+      result = mathHelper.add(leftNum, rightNum)
+      outputString = result.toString()
+      updateOutPutString()
+   }
```
Comment (cwbowron) [inline app/src/main/java/com/example/kharrison/my_application/CalculatorController.kt:127]: There's a lot of repeated code in `add`, `subtract`, etc. You might consider moving the common code out of the individual functions. 

    when (selectedOperator) {
         ADDITION -> restult = mathHelper.add(leftNum, rightNum)
         ...
    }
    
    outputString = result.toString()
    updateOutPutString()

    ....

**Comment by cwbowron** [comment]
:+1:

**Review by cwbowron — APPROVED** [review]

File: app/src/main/java/com/example/kharrison/my_application/CalculatorController.kt (hunk 3 above)
Comment (Kevin-J-Harrison) [reply app/src/main/java/com/example/kharrison/my_application/CalculatorController.kt:127]: That's actually how it was initially set up. Bill and I pulled out the Division function to handle dividing by zero, and did the others as well if any special use cases arose for them. It's definitely a spot for simplification though.

**Pull request closed without merging**